### PR TITLE
A basic implementation of feasibility based bounds tightening

### PR DESCRIPTION
**Pull request by michaelbynum**
## Summary/Motivation:
A basic implementation of feasibility based bounds tightening.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/fbbt/tests/test_fbbt.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+try:
+    import numpy as np
+    numpy_available = True
+except ImportError:
+    numpy_available = False
+
+
+class TestFBBT(unittest.TestCase):
+    def test_add(self):
+        if not numpy_available:
```
Comment (blnicho) [inline pyomo/contrib/fbbt/tests/test_fbbt.py]: I think this if statement should be promoted to a `unittest.skipIf` decorator on the test method for consistency with other Pyomo tests

File: pyomo/contrib/fbbt/tests/test_interval.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    import numpy as np
+    numpy_available = True
+    np.random.seed(0)
+except ImportError:
+    numpy_available = False
+
+
+class TestInterval(unittest.TestCase):
+    def test_add(self):
+        if not numpy_available:
```
Comment (blnicho) [inline pyomo/contrib/fbbt/tests/test_interval.py]: Same as above, I suggest using decorators instead of if statements

**Comment by michaelbynum** [comment]
Thanks for the feedback @blnicho. These are good suggestions.

**Comment by michaelbynum** [comment]
Appveyor and codecov did not run the last time I pushed, so I merged in master and pushed again. Hopefully all of the tests will run this time.

**Comment by michaelbynum** [comment]
@carldlaird I believe I addressed your concerns. Can you please take another look?

**Pull request merged**